doc/lua_api.txt

```diff
@@ -5470,8 +5470,14 @@ This is basically a reference to a C++ `ServerActiveObject`
     * `acc` is a vector
 * `get_acceleration()`: returns the acceleration, a vector
 * `set_rotation(rot)`
-    * `rot` is a vector (radians). X is pitch (elevation), Y is yaw (heading)
-      and Z is roll (bank).
+    * `rot` is a vector (radians).
+    * `rot.y` is rotation around the world Y axis, and acts like `set_yaw()`.
+    * `rot.x` is rotation around the object's left-right axis, so can be considered
```

Comment:
It'd be nicer to list the components in alphabetical order. i.e. x, y, z

Reply:
The order is intentional and explained in the issue https://github.com/minetest/minetest/issues/8868#issuecomment-526355920
How rotation works is far clearer when understood in the order YXZ, because that's the order of the application of the rotation components, and the order of what axes are rotated by what other axes.
I won't change this order as it's logical and essential for clarity

Y coming first is also consistent with how previously we only had 'set yaw', and how the effect of yaw is unchanged and has priority.

Reply:
Order of application of rotation deserves to explicitly mentioned, IMO.

Reply:
It is, although maybe not as clearly as it could be:
```
The object's left-right axis is rotated around the world Y axis by rot.y
The object's front-back axis is rotated around the object's left-right axis
```

Reply:
No, I meant that the order of application of rotation should be mentioned as a separate point, and the components should be documented in alphabetical order. Modders are used to X first, then Y and Z, and it'd be more intuitive for them if the documentation lists them in correct order. And as you said, the ordering of the lines doesn't offer any clue as to what the order of rotation is.

Reply:
I need to improve this (see below).
The explanation should hopefully explain what you mean by 'order', it may be clearer if it is not a separate point but integrated into the explanation, as the current form of the PR attempts to.
> And as you said, the ordering of the lines doesn't offer any clue as to what the order of rotation is.

I didn't write that, i actually wrote the opposite, see "it is" above, i just wrote that it could be clearer.
I also made it clear that the YXZ component order of the documentation is specifically intended to explain the 'order of rotation'.
I'm bemused why you would claim i wrote something that is so clearly opposite to what i actually wrote.

I've also explained why the YXZ component order of the documentation is essential, and given several good reasons. Ordering for the sake of being alphabetical is a non-argument. We're used to XYZ elsewhere because it makes sense elsewhere, but here it doesn't. Here, YXZ will be more intuitive for those trying to understand these rotations.

Reply:
> I didn't write that, i actually wrote the opposite, see "it is" above, i just wrote that it could be clearer.

That it could be made clearer is what I was referring to, sorry for the confusion. :)

> I also made it clear that the YXZ component order of the documentation is specifically intended to explain the 'order of rotation'.

The order of documentation is **implicit**, and isn't very helpful in indicating the order of rotation. As a modder, the first thing that'd come to my mind when I look at this documentation is that the order is messed up, and I would not have any way of knowing that the order of documentation of the components corresponds to the order of application of rotation. This is because there's currently nothing in the documentation that indicates that the points are documented in a "sequence".

Inclusion of words/phrases like "First, the object is rotated in the ... axis", and "Then the object is rotated in the ... axis" would indicate the existence of a sequence. But then the documentation would become unnecessarily verbose, which is why I suggested **explicitly** mentioning the order of application of rotations. e.g.

```
  * Note: Rotations are applied in the following order: Y -> X -> Z
  * <Document X> 
  * <Document Z> 
  * <Document Y>
```